### PR TITLE
Add v2 endpoints

### PR DESCRIPTION
**Pull request by JacobPlaster**
### Description:
This pull request adds all of the new v2 rest endpoints and adds the capability to specify leverage when creating an order.

 ### Breaking changes:
-None

 ### New features:
- [x] Adds function rest/v2 submit_order
- [x] Adds function rest/v2 update_order
- [x] Adds function rest/v2 cancel_order
- [x] Adds function rest/v2 submit_funding_offer
- [x] Adds function rest/v2 cancel_funding_offer
- [x] Adds function rest/v2 close_funding
- [x] Adds function rest/v2 submit_funding_auto
- [x] Adds function rest/v2 transfer
- [x] Adds function rest/v2 deposit_address
- [x] Adds function rest/v2 create_deposit_address
- [x] Adds function rest/v2 withdraw
- [x] Adds function rest/v2 claim_position
- [x] Updates order model to allow lev field
- [x] Updates position model to accept more data fields

 ### Fixes:
Fix fields for Order model, originally it was attempting to get some fields (like hidden) using the wrong index.

 ### PR status:
- [x] Version bumped
- [x] Change-log updated
- ~~Documentation updated~~ - no documentation for rest yet

### CODE REVIEW

File: lib/models/order.rb [hunk 1]
```diff
@@ -16,14 +16,28 @@ class Order < Model
...
         :price_avg => 17,
         :price_trailing => 18,
         :price_aux_limit => 19,
-        :notify => 23,
-        :placed_id => 25
+        # placeholder
+        # placeholder
+        # placeholder
+        # placeholder
+        :notify => 24,
```
Comment (f3rno) [inline lib/models/order.rb]: This doesn't seem correct, it is at index 23 in the docs & `bfx-api-node-models`, can you confirm?

File: lib/rest/v2/funding.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      elsif funding.kind_of?(Hash)
+        id = funding[:id] || order['id']
+      else
+        raise Exception, 'tried to close funding with invalid ID'
+      end
+      authenticated_post("auth/w/funding/close", params: { :id => id }).body
+    end
+
+    ###
+    # Close a funding loan/credit
```
Comment (f3rno) [inline lib/rest/v2/funding.rb]: Func doc header incorrect

File: lib/rest/v2/orders.rb [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    ###
+    # Update an order with a changeset by ID
+    #
+    # @param [Hash] changes - must contain ID
+    #
+    # @return [Array] Raw notification
+    ###
+    def update_order (changes)
+      id = changes[:id] || changes['id']
```
Comment (f3rno) [inline lib/rest/v2/orders.rb]: Unused var

File: lib/rest/v2/positions.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      if position.is_a?(Numeric)
+        id = position
+      elsif position.is_a?(Array)
+        id = position[0]
+      elsif position.instance_of?(Models::Position)
+        id = position.id
+      elsif position.kind_of?(Hash)
+        id = position[:id] || position['id']
+      else
+        raise Exception, 'tried to cancel position with invalid ID'
```
Comment (f3rno) [inline lib/rest/v2/positions.rb]: `cancel` -> `claim`

File: lib/models/order.rb (hunk 1 above)
Comment (JacobPlaster) [reply lib/models/order.rb]: according to the docs:

```
[
  #    ID, 1
  #    GID, 2
  #    CID, 3
  #    SYMBOL, 4
  #    MTS_CREATE, 5 
  #    MTS_UPDATE, 6
  #    AMOUNT, 7
  #    AMOUNT_ORIG, 8 
  #    TYPE, 9
  #    TYPE_PREV, 10
  #    MTS_TIF, 11
  #    _PLACEHOLDER, 12
  #    FLAGS, 13
  #    ORDER_STATUS, 14
  #    _PLACEHOLDER, 15
  #    _PLACEHOLDER, 16
  #    PRICE, 17
  #    PRICE_AVG, 18
  #    PRICE_TRAILING, 19
  #    PRICE_AUX_LIMIT, 20
  #    _PLACEHOLDER, 21
  #    _PLACEHOLDER, 22
  #    _PLACEHOLDER, 23
  #    _PLACEHOLDER, 24
  #    HIDDEN, 25
  #    PLACED_ID, 26
  #    _PLACEHOLDER, 27
  #    _PLACEHOLDER, 28
  #    ROUTING, 29
  #    _PLACEHOLDER, 30
  #    _PLACEHOLDER, 31
  #    META 32
  #  ],
```

Comment (JacobPlaster) [reply lib/models/order.rb]: so by that, hidden (notify) is 24 (I forgot to zero index)

Comment (JacobPlaster) [reply lib/models/order.rb]: Ignore this - turns out the docs is missing the `NOTIFY` field in the v2 rest response which confused met. Fixed now

**Pull request merged**